### PR TITLE
Mail: Ask the user if they want to see the help man page

### PR DESCRIPTION
**Pull request by Undefine-Unity**
If there are no mail server configured, then the application will ask the
user, if they want to see the help man page.

### CODE REVIEW

File: Userland/Applications/Mail/MailWidget.cpp [hunk 1]
```diff
@@ -103,7 +103,10 @@ bool MailWidget::connect_and_login()
     auto server = Config::read_string("Mail", "Connection", "Server", {});
 
     if (server.is_empty()) {
-        GUI::MessageBox::show_error(window(), "Mail has no servers configured. Refer to the Mail(1) man page for more information.");
+        int result = GUI::MessageBox::show(window(), "Mail has no servers configured. Do you want to see the Mail(1) man page for more information?", "Error", GUI::MessageBox::Type::Error, GUI::MessageBox::InputType::YesNo);
+        if (result == GUI::MessageBox::ExecResult::ExecYes) {
+            Desktop::Launcher::open(URL::create_with_file_protocol("/usr/share/man/man1/Mail.md"), "/bin/Help");
+        }
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/Mail/MailWidget.cpp:109]: See the code style; don't use braces for single-line ifs.

File: Userland/Applications/Mail/main.cpp [hunk 2]
```diff
@@ -45,6 +45,11 @@ int main(int argc, char** argv)
         return 1;
     }
 
+    if (unveil("/tmp/portal/launch", "rw") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/Mail/main.cpp:52]: Why is this change in this commit? Are you fixing an issue with unveil errors? Does that relate to the functional change the commit states it makes?

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]

File: Userland/Applications/Mail/main.cpp (hunk 2 above)
Comment (trflynn89) [reply Userland/Applications/Mail/main.cpp:52]: `Desktop::Launcher` requires this.

Comment (Undefine-Unity) [reply Userland/Applications/Mail/main.cpp:52]: Yes, it's required by `Desktop::Launcher`.

Comment (kleinesfilmroellchen) [reply Userland/Applications/Mail/main.cpp:52]: Then do that as a separate fix beforehand, or specify why this change is done in the commit message.

Comment (trflynn89) [reply Userland/Applications/Mail/main.cpp:52]: > Then do that as a separate fix beforehand

It's not a fix. It belongs in the same commit and IMO is obvious enough to not require further details.

**Comment by awesomekling** [comment]
Hmm, how about we open the `MailSettings` app directly, instead of opening a man page that eventually tells you to edit the mail settings? :^)

**Comment by kleinesfilmroellchen** [comment]
@awesomekling now I feel dumb (^:

**Comment by bcoles** [comment]
Traditionally, mail applications have a sizable attack surface (processing arbitrary rich mail content from arbitrary remote hosts). Although this is somewhat mitigated on Serenity through the use of sand boxing for HTML rendering and image processing, which is great.

Given that the Desktop Launcher is designed to launch arbitrary applications with arbitrary arguments, it would be nice to lock this down so that any compromise of the Mail application leads only to compromise of mail content and does not permit execution of arbitrary programs.

To achieve this, Desktop Launcher can be locked down using a similar mechanism to `unveil`.

https://github.com/SerenityOS/serenity/blob/e9f66d1c2aa959d8f0a62fc23a6cf01a002d01f0/Userland/Libraries/LibDesktop/Launcher.cpp#L54-L82

Here's an example from `ImageViewer` that locks down the launcher to opening only the Help application and only with the man page as an argument, then locks the allowed list to prevent further modification.

https://github.com/SerenityOS/serenity/blob/e9f66d1c2aa959d8f0a62fc23a6cf01a002d01f0/Userland/Applications/ImageViewer/main.cpp#L48-L57

You could take a similar approach to whitelisting the `MailSettings` application only.

**Comment by kleinesfilmroellchen** [comment]
@Undefine-Unity do you want to get this merged still?

**Comment by Undefine-Unity** [comment]
I will update this pr this weekend

**Comment by kleinesfilmroellchen** [comment]
Hint: rebase on master (`git checkout mail_manpage; git fetch origin/master; git rebase origin/master mail_manpage`) to resolve the merge conflicts.

**Pull request merged**